### PR TITLE
Fix 'video.tools.drawing.color_gradient'

### LINKED ISSUE #1256: “The End” effect example scripts get fault -- TypeError: 'NoneType' object is not subscriptable
I run the example scripts “The End” effect  after complete the installation,the code is:
```
from moviepy.editor import *
from moviepy.video.tools.drawing import circle

clip = VideoFileClip("myvideo.mp4", audio=False).\
           subclip(26,31).\
           add_mask()
           
w,h = clip.size

# The mask is a circle with vanishing radius r(t) = 800-200*t               
clip.mask.get_frame = lambda t: circle(screensize=(clip.w,clip.h),
                                       center=(clip.w/2,clip.h/4),
                                       radius=max(0,int(800-200*t)),
                                       col1=1, col2=0, blur=4)


the_end = TextClip("The End", font="Amiri-bold", color="white",
                   fontsize=70).set_duration(clip.duration)

final = CompositeVideoClip([the_end.set_pos('center'),clip],
                           size =clip.size)
                           
final.write_videofile("theEnd.mp4")
```

#### Expected Behavior

Return correct

#### Actual Behavior

```
Traceback (most recent call last):
  File "theend.py", line 34, in <module>
    video.write_videofile("theEnd.mp4")
  File "<decorator-gen-61>", line 2, in write_videofile
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/decorators.py", line 56, in requires_duration
    return f(clip, *a, **k)
  File "<decorator-gen-60>", line 2, in write_videofile
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/decorators.py", line 135, in use_clip_fps_by_default
    return f(clip, *new_a, **new_kw)
  File "<decorator-gen-59>", line 2, in write_videofile
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/decorators.py", line 24, in convert_masks_to_RGB
    return f(clip, *a, **k)
  File "<decorator-gen-58>", line 2, in write_videofile
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/decorators.py", line 89, in wrapper
    return f(*new_a, **new_kw)
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/video/VideoClip.py", line 350, in write_videofile
    logger=logger,
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/video/io/ffmpeg_writer.py", line 245, in ffmpeg_write_video
    logger=logger, with_times=True, fps=fps, dtype="uint8"
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/Clip.py", line 473, in iter_frames
    frame = self.get_frame(t)
  File "<decorator-gen-11>", line 2, in get_frame
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/decorators.py", line 89, in wrapper
    return f(*new_a, **new_kw)
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/Clip.py", line 98, in get_frame
    return self.make_frame(t)
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/video/compositing/CompositeVideoClip.py", line 123, in make_frame
    f = c.blit_on(f, t)
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/video/VideoClip.py", line 599, in blit_on
    mask = self.mask.get_frame(ct) if self.mask else None
  File "movie.py", line 15, in <lambda>
    col1=1, col2=0,blur=4,)
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/video/tools/drawing.py", line 280, in circle
    offset=offset,
  File "/home/pandahub/.local/lib/python3.6/site-packages/moviepy/video/tools/drawing.py", line 146, in color_gradient
    vector = np.array(vector[::-1])
TypeError: 'NoneType' object is not subscriptable
```

#### Steps to Reproduce the Problem
1. install the moviepy ( I have tried both 1.0.3 & v2.0.0.dev1 version, and both are fault)
   pip install moviepy
2. install the ffmpeg, or it will be fault with "ffmpeg not installed" when run the script
  sudo apt update
  sudo add-apt-repository ppa:jonathonf/ffmpeg-4
  sudo apt install ffmpeg
3. update the /etc/ImageMagick-6/policy.xml, or it will be fault with "ImageMagick not installed" when run the script
  <policy domain="path" rights="none" pattern="@*" />
   changed to 
  
4. finally I run the script, and it get thta error message: TypeError: 'NoneType' object is not subscriptable
  python theend.py

#### Specifications

  - Python Version: python3.6
  - Moviepy Version: both 1.0.3 & v2.0.0.dev1
  - Platform Name: ubuntu 18.04
  - Platform Version: 18.04

### PR DESCRIPTION
**Pull request by mondeja**
Closes #1256

### CODE REVIEW

**Comment by tburrows13** [comment]
I think that this still needs a bigger rework? The issue is that `circle()` doesn't provide `p2` or `vector`, and I think that the code will still fail in this case.

File: moviepy/video/tools/drawing.py [hunk 1]
```diff
@@ -124,19 +124,21 @@ def color_gradient(
...
     if shape == "linear":
+        if vector is None:
+            if p2 is not None:
+                p2 = np.array(p2[::-1])
+                vector = p2 - p1
+            else:
+                raise ValueError("You must provide either 'p2' or 'vector'")
+        else:
+            vector = np.array(vector[::-1])
+            p2 = p1 + vector
```
Comment (tburrows13) [inline moviepy/video/tools/drawing.py:138]: This line isn't needed

**Review by YannickLehr — APPROVED** [review]
fix works

**Comment by mondeja** [comment]
I've done tests with all possible combinations and I'm sure that now works and is more efficient, so I'm merging...

**Pull request merged**